### PR TITLE
Stable tanh bijector

### LINKED ISSUE #455: TTP: stable tanh

### PR DESCRIPTION
**Pull request by Haichao-Zhang**

### CODE REVIEW

File: alf/utils/dist_utils.py [hunk 1]
```diff
@@ -24,6 +24,48 @@
...
+
+    def _inverse(self, y):
+        # Based on https://github.com/tensorflow/agents/commit/dfb8c85a01d65832b05315928c010336df13f7b9#diff-a572e559b953f965c5c2cd1b9ded2c7b
+
+        # 0.99999997 is the maximum value such that atanh(x) is valid for both float32 and float64
+        def atanh(x):
+            return 0.5 * torch.log((1 + x) / (1 - x))
+
+        y = torch.where(
+            torch.abs(y) <= 1.0, torch.clamp(y, -0.99999997, 0.99999997), y)
```
Comment (hnyu) [inline alf/utils/dist_utils.py:61]: Shouldn't abs(y) <= 1.0 always be true?

File: alf/utils/dist_utils.py [hunk 2]
```diff
@@ -24,6 +24,48 @@
...
+        return atanh(y)
+
+    def log_abs_det_jacobian(self, x, y):
+        return 2.0 * (torch.log(
+            torch.tensor(2.0, dtype=x.dtype, requires_grad=False).detach()) - x
```
Comment (hnyu) [inline alf/utils/dist_utils.py]: detach is unnecessary if requires_grad=False?

File: alf/utils/dist_utils.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/utils/dist_utils.py:61]: As an inverse function ```abs(y) <= 1``` should be true. As a general function ```y``` could be a general input with ```abs(y) > 1```.  In this case, for the invalid ```y``` values (```abs(y)>1```), we keep them unchanged and won't clamp them into valid range.

File: alf/utils/dist_utils.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/utils/dist_utils.py]: detach removed

**Review by hnyu — APPROVED** [review]
